server/akka-http-server/src/test/scala/sttp/tapir/server/akkahttp/AkkaHttpServerTest.scala

```diff
@@ -43,7 +37,7 @@ class AkkaHttpServerTest extends TestSuite with EitherValues {
       implicit val m: FutureMonad = new FutureMonad()(actorSystem.dispatcher)
 
       val interpreter = new AkkaHttpTestServerInterpreter()(actorSystem)
-      val createServerTest = new CreateServerTest(interpreter)
+      val createServerTest = new DefaultCreateServerTest(backend, interpreter).asInstanceOf[DefaultCreateServerTest[Future, AkkaStreams with WebSockets, Route, AkkaResponseBody]]
```

Comment:
I suspect the `.asInstanceOf` is a type inference issue - maybe it would help if we explicitly added type parameters?

Reply:
The `backend` has `FS2Streams` as capability and here we require `AkkaStreams` for `ServerWebSocketTests` and `ServerStreamingTests`, so `.asInstanceOf` was the only option

Reply:
and this works? if the streams are incompatible, it cannot possibly work

Reply:
ah I think I see where the problem is. It's wrong to require that the `R` of the backend is the same as of the interpreter. The backend's streaming can always be Fs2? as that's what we are using in tests.

Reply:
why did we need to parametrise with the backend in the first place? I know there was a reason but forgot what it was ;)

Reply:
To be able to run tests like `ServerBasicTests` with stubbed backend (`AwsLambdaCreateServerStubTest`). While doing that each test needs a new instance of `StubBackend`.